core/products/future.go

```diff
@@ -243,13 +258,28 @@ func NewFuture(ctx context.Context, log *logging.Logger, f *types.Future, oe Ora
 		return nil, err
 	}
 
+	dSrcSpec := f.DataSourceSpecForSettlementData.Data.GetDataSourceSpecConfiguration()
+	if dSrcSpec != nil {
+		for _, f := range dSrcSpec.Filters {
+			// Oracle specs with more than one unique filter names are not allowed to exists, so we do not have to make that check here.
+			// We are good to only check if the type is `PropertyKey_TYPE_DECIMAL` or `PropertyKey_TYPE_INTEGER`, because we take decimals
+			// into consideration only in those cases.
+			if f.Key.Type == datapb.PropertyKey_TYPE_INTEGER && f.Key.NumberDecimalPlaces != nil {
+				oracleBinding.settlementDataPropertyType = f.Key.Type
+				if f.Key.Type == datapb.PropertyKey_TYPE_INTEGER {
```

Comment:
We've already checked the key type in the outer `if`, this can be removed here

Reply:
Yes, this is fixed. Thanks